src/libs/actions/PersonalDetails.js

```diff
@@ -262,6 +262,39 @@ function setPersonalDetails(details, shouldGrowl) {
         });
 }
 
+function updateProfile(firstName, lastName, pronouns, timezone) {
+    const myPersonalDetails = personalDetails[currentUserEmail];
+    API.write('UpdateProfile', {
+        details: JSON.stringify({firstName, lastName, pronouns}),
+        timezone: JSON.stringify(timezone),
+    }, {
+        optimisticData: [{
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS,
+            value: {
+                [currentUserEmail]: {
+                    firstName,
+                    lastName,
+                    pronouns,
+                    timezone,
+                },
+            },
+        }],
```

Comment:
@Beamanator should this optimistic data actually update it in the UI automatically? Is it that we do not rerender the RHS when we go back from the settings? 

In profile settings:
<img width="375" alt="image" src="https://user-images.githubusercontent.com/36083550/181575916-3a549159-bc44-475d-9706-2a5969ebb6c6.png">

Clicking back to the settings, still old name 
<img width="376" alt="image" src="https://user-images.githubusercontent.com/36083550/181575986-b170b9c6-c208-4f77-96ae-29e07506fc70.png">


I think that should be fixed because as user I would be perplexed what is correct.